lib/linter.js

```diff
@@ -405,26 +405,28 @@ export default class Linter {
       return currentSource;
     }
 
+    let filePath = options.filePath;
     let fileConfig = this._moduleStatusCache.getConfigForFile(options);
     let ruleNames = new Set(fixableIssues.map((r) => r.rule));
 
-    if (isGlimmerFileExtension(options.filePath)) {
-      return await this.#byType.glimmerScript.fix(ruleNames, options, fileConfig);
-    }
-
-    if (isHBS(options.filePath) || isHTML(options.filePath)) {
-      return this.#byType.template.fix(ruleNames, options, fileConfig);
-    }
-
-    if (isHbsTemplateInScriptCheckingEnabled(options) && isScript(options.filePath)) {
-      return await this.#byType.script.fix(ruleNames, options, fileConfig);
-    }
+    return await processFile(filePath, {
```

Comment:
The two sets of `if`-statements (before) has the risk of getting out-of-sync. The utility `processFile()` helps us apply the same order (while hiding details).

Reply:
Thank you!